src/Controller/CourseController.php

```diff
@@ -48,21 +66,43 @@ public function setCourse(Request $request)
             return new JsonResponse([
                  'error_message' => $errors
              ], Response::HTTP_BAD_REQUEST);
+
         }
 
         $courseUser = new CourseUser();
         $courseUser->setUser($this->getUser());
         $courseUser->setRole(RoleInterface::ADMIN);
         $courseUser->setStatus(StatusInterface::ACTIVE);
 
+        if($course->getIsSubmittable()){
+            $entryTask = new EntryTask();
+            $entryTask->setDescription($data['entry_task_description']);
+            $entryTask->setDeadlineDate($data['entry_task_deadline_date']);
+
+            $errors = $this->validator->validate($entryTask);
+
+            if (count($errors) > 0) {
+
+                return new JsonResponse([
+                    'error_message' => $errors->get(0)->getMessage(),
+                ], Response::HTTP_BAD_REQUEST);
+            }
+        }
+
         try {
             $em = $this->getDoctrine()->getManager();
             $em->persist($course);
             $em->flush();
 
+            if(isset($entryTask)){
+                $entryTask->setCourse($course);
+                $em->persist($entryTask);
+            }
+
```

Comment:
Čia neturėtum daryti atskirai set cource. Tavo `$cource->setUser($user);` jau turėtų kviesti `$user->setCource($this);`. Ir formoje, kuri turi tą fieldą reikia nurodyt specialų parametrą, kad naudotų setterius, o ne per reflection'ą sudėtų nepamenu tiksliai kaip tas parametras vadinas, pasigooglink, tikrai rasi nesunkiai :)